### DISCUSSION TITLE
How to use external odometry as motion prediction in ICP odometry.

### DISCUSSION THREAD

**Discussion by zafaryab-a** [introlab/rtabmap#1035] [How to]
I want to enable using external odometry (Wheel Encoder) as motion prediction in ICP odometry. I am unable to find any parameter for same.

**Comment by matlabbe** [comment]
You should use `guess_frame_id` parameter of `icp_odometry` node to use TF from your wheel odometry. Your wheel odometry will then be used as motion prediction, otherwise constant velocity model is used.
